models/catalog.cattle.io.app.js

```diff
@@ -118,6 +115,39 @@ export default class CatalogApp extends SteveModel {
     return sortable(version);
   }
 
+  get currentVersionCompatible() {
+    const workerOSs = this.$rootGetters['currentCluster'].workerOSs;
+
+    const chart = this.matchingChart(false);
+    const thisVersion = this.spec?.chart?.metadata?.version;
+
+    if (!chart) {
+      return true;
+    }
+
+    const versionInChart = chart.versions.find(version => version.version === thisVersion);
+
+    if (!versionInChart) {
+      return true;
+    }
+    const compatibleVersions = compatibleVersionsFor(chart, workerOSs, true) || [];
+
+    const thisVersionCompatible = !!compatibleVersions.find(version => version.version === thisVersion);
+
+    return thisVersionCompatible;
+  }
+
+  get stateDescription() {
+    if (this.currentVersionCompatible) {
+      return null;
+    }
+    if (this.upgradeAvailable) {
+      return this.t('catalog.os.versionIncompatible');
```

Comment:
This seems redundant, is there a plan to add a third stateDescription at some point in the future?